lib/api/src/native.rs

```diff
@@ -115,98 +109,89 @@ macro_rules! impl_native_traits {
+                    // of the correct size here.
+                    let mut params_list = [ $( $x.to_native().to_binary() ),* ];
+                    let mut rets_list_array = Rets::empty_array();
+                    let rets_list = rets_list_array.as_mut();
+                    let using_rets_array;
+                    let args_rets: &mut [i128] = if params_list.len() > rets_list.len() {
+                        using_rets_array = false;
+                        params_list.as_mut()
+                    } else {
+                        using_rets_array = true;
+                        for (i, &arg) in params_list.iter().enumerate() {
+                            rets_list[i] = arg;
+                        }
+                        rets_list.as_mut()
+                    };
+                    unsafe {
+                        wasmer_vm::wasmer_call_trampoline(
+                            self.vmctx(),
+                            trampoline,
+                            self.address(),
+                            args_rets.as_mut_ptr() as *mut u8,
+                        )
+                    }?;
+                    let num_rets = rets_list.len();
+                    if !using_rets_array && num_rets > 0 {
+                        let src_pointer = params_list.as_ptr();
+                        let rets_list = &mut rets_list_array.as_mut()[0] as *mut i128;
                         unsafe {
-                            wasmer_vm::wasmer_call_trampoline(
-                                self.vmctx(),
-                                trampoline,
-                                self.address(),
-                                args_rets.as_mut_ptr() as *mut u8,
-                            )
-                        }?;
-                        let num_rets = rets_list.len();
-                        if !using_rets_array && num_rets > 0 {
-                            let src_pointer = params_list.as_ptr();
-                            let rets_list = &mut rets_list_array.as_mut()[0] as *mut i128;
-                            unsafe {
-                                // TODO: we can probably remove this copy by doing some clever `transmute`s.
-                                // we know it's not overlapping because `using_rets_array` is false
-                                std::ptr::copy_nonoverlapping(src_pointer,
-                                                              rets_list,
-                                                              num_rets);
-                            }
+                            // TODO: we can probably remove this copy by doing some clever `transmute`s.
+                            // we know it's not overlapping because `using_rets_array` is false
+                            std::ptr::copy_nonoverlapping(src_pointer,
+                                                            rets_list,
+                                                            num_rets);
```

Comment:
Indentation looks wrong? Does this pass lint?

Reply:
Yeah, weird. This change was made by the linter

Reply:
The body of macros are not formatted by rustfmt. If the formatting looks wrong, it has to be fixed manually. It was likely caused by your editor if you didn't do it